evm-contracts/src/v0.6/examples/VRFD20.sol

```diff
@@ -0,0 +1,164 @@
+     *
+     * @param userProvidedSeed uint256 unpredictable seed
+     * @param roller address of the roller
+     */
+    function rollDice(uint256 userProvidedSeed, address roller) public onlyOwner returns (bytes32 requestId) {
+        require(LINK.balanceOf(address(this)) >= s_fee, "Not enough LINK to pay fee");
+        require(s_results[roller] == 0, "Already rolled");
+        requestId = requestRandomness(s_keyHash, s_fee, userProvidedSeed);
+        s_rollers[requestId] = roller;
+        s_results[roller] = ROLL_IN_PROGRESS;
+        emit DiceRolled(requestId, roller);
+    }
+
+    /**
+     * @notice Callback function used by VRF Coordinator to return the random number
+     * to this contract.
+     * @dev This is where you do something with randomness!
+     * @dev The VRF Coordinator will only send this function verified responses.
+     *
+     * @param requestId bytes32
+     * @param randomness The random result returned by the oracle
+     */
+    function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
+        uint256 result = randomness.mod(20).add(1);
+        s_results[s_rollers[requestId]] = result;
+        emit DiceLanded(requestId, result);
+    }
+
+    /**
+     * @notice Get the house assigned to the player once the address has rolled
+     * @param player address
+     * @return house as a string
+     */
+    function house(address player) public view returns (string memory) {
+        require(s_results[player] != 0, "Dice not rolled");
+        require(s_results[player] != ROLL_IN_PROGRESS, "Roll in progress");
+        return getHouseName(s_results[player]);
+    }
+
+    /**
+     * @notice Withdraw LINK from this contract.
+     * @param to the address to withdraw LINK to
+     * @param value the amount of LINK to withdraw
+     */
+    function withdrawLINK(address to, uint256 value) public onlyOwner {
+        require(LINK.transfer(to, value), "Not enough LINK");
+    }
+
+    /**
+     * @notice Set the key hash for the oracle
+     *
+     * @param keyHash bytes32
+     */
+    function setKeyHash(bytes32 keyHash) public onlyOwner {
+        s_keyHash = keyHash;
+    }
+
+    /**
+     * @notice Get the current key hash
+     *
+     * @return bytes32
+     */
+    function keyHash() public view returns (bytes32) {
+        return s_keyHash;
+    }
+
+    /**
+     * @notice Set the oracle fee for requesting randomness
+     *
+     * @param fee uint256
+     */
+    function setFee(uint256 fee) public onlyOwner {
+        s_fee = fee;
+    }
+
+    /**
+     * @notice Get the current fee
+     *
+     * @return uint256
+     */
+    function fee() public view returns (uint256) {
+        return s_fee;
+    }
+
+    /**
+     * @notice Get the house namne from the id
+     * @param id uint256
+     * @return house name string
+     */
+    function getHouseName(uint256 id) private view returns (string memory) {
+        string[20] memory houseNames = [
```

Comment:
This disregards the Free Folk, who have no house. We're going to need you to apologize on #gameofthrones Twitter.😊

Reply:
The savages have no place in Westeros